File: api/FinanceApi/FinanceApi/Repositories/EF Models/vExpenseDetails.cs
using System.ComponentModel.DataAnnotations;

namespace FinanceApi.Repositories.EF_Models
{
    public class vExpenseDetails
    {
        [Key]
        public int ExpenseID { get; set; }
        public required string ExpenseTypeName { get; set; }
        public required string PaymentTypeName { get; set; }
        public required string PaymentTypeCategoryName { get; set; }
        public bool IsIncome { get; set; }
        public bool IsInvestment { get; set; }
        public int ExpenseTypeID { get; set; }
        public int PaymentTypeID { get; set; }
        public required string PaymentTypeDescription { get; set; }
        public int PaymentTypeCategoryID { get; set; }
        public DateTime ExpenseDate { get; set; }
        public DateTime LastUpdated { get; set; }
        public decimal ExpenseAmount { get; set; }
    }
}

File: api/Node/finance-api/src/entity/v_expense_details.ts
import { ViewEntity, ViewColumn} from "typeorm";

@ViewEntity({
	name: "v_expense_detail",
	expression: '',
	synchronize: false
})
export class v_expense_detail {
  @ViewColumn({name: 'expense_id'})
  expenseID?: number;

  @ViewColumn({name: 'expense_type_id'})
  expenseTypeID?: number;

  @ViewColumn({name: 'payment_type_id'})
  paymentTypeID?: number;

  @ViewColumn({name: 'payment_type_category_id'})
  paymentTypeCategoryID?: number;

  @ViewColumn({name: 'expense_type_name'})
  expenseTypeName?: string;

  @ViewColumn({name: 'payment_type_name'})
  paymentTypeName?: string;

  @ViewColumn({name: 'payment_type_description'})
  paymentTypeDescription?: string;

  @ViewColumn({name: 'payment_type_category_name'})
  paymentTypeCategoryName?: string;

  @ViewColumn({name: 'is_income'})
  isIncome?: boolean;

  @ViewColumn({name: 'is_investment'})
  isInvestment?: boolean;

  @ViewColumn({name: 'expense_description'})
  expenseDescription?: string;

  @ViewColumn({name: 'expense_amount'})
  expenseAmount?: number;

  @ViewColumn({name: 'expense_date'})
  expenseDate?: Date;

  // @ViewColumn()
  // LastUpdated?: Date;
}
